tools/x-docs-2/src/components/sidebar/module-list.jsx

```diff
@@ -29,15 +31,15 @@ const buildSubheadingsList = (menu, currentNode) => {
 	});
 }
 
-const Sidebar = ({ data, title, menu }) => (
+const Sidebar = ({ data, title, menu, pageContext }) => (
 	<ul className="site-sidebar">
-		<li className="site-sidebar__item site-sidebar__item--title">{title}</li>
-		{data.map(({ node }, i) => (
+		<li className="site-sidebar__item site-sidebar__item--title">{title || pageContext.source}</li>
+		{ data.map(({ node }, i) => (
 			<li key={`list-${i}`} className="site-sidebar__item site-sidebar__item--link">
 				<Link to={node.path} exact activeClassName="is-active">
 					{node.context.title}
 				</Link>
-				{ menu && node.path === document.location.pathname ? (
+				{ menu && menu.headings.length > 0 && pageContext && pageContext.source && pageContext.title && node.path === `/${pageContext.source}/${pageContext.title}` ? (
```

Comment:
If we were to change the logic we use to build slugs this would break. Instead of re-constructing the slug here see if you get access to the `location` prop passed to the page instead =]

Reply:
That makes sense =)